c/tskit/tables.c

```diff
@@ -8172,16 +8172,69 @@ tsk_check_subset_equality(tsk_table_collection_t *self, tsk_table_collection_t *
     return ret;
 }
 
+static int
+tsk_table_collection_add_and_remap_node(tsk_table_collection_t *self, tsk_table_collection_t *other, tsk_id_t k, tsk_id_t *individual_map, tsk_id_t *population_map, tsk_id_t *node_map, bool add_populations) {
+    int ret = 0;
+    tsk_id_t new_ind, new_pop;
+    tsk_node_t node;
+    tsk_individual_t ind;
+    tsk_population_t pop;
+    
+    ret = tsk_node_table_get_row(&other->nodes, k, &node);
+    if (ret < 0) {
+        goto out;
+    }
+    new_ind = TSK_NULL;
+    if (node.individual != TSK_NULL) {
+        if (individual_map[node.individual] == TSK_NULL) {
+            tsk_individual_table_get_row(
+                &other->individuals, node.individual, &ind);
+            ret = tsk_individual_table_add_row(&self->individuals, ind.flags,
+                ind.location, ind.location_length, ind.metadata,
+                ind.metadata_length);
+            if (ret < 0) {
+                goto out;
+            }
+            individual_map[node.individual] = ret;
+        }
+        new_ind = individual_map[node.individual];
+    }
+    new_pop = TSK_NULL;
+    if (node.population != TSK_NULL) {
+        // keep same pops if add_populations is False
+        if (!add_populations) {
+            population_map[node.population] = node.population;
+        }
+        if (population_map[node.population] == TSK_NULL) {
+            tsk_population_table_get_row(
+                &other->populations, node.population, &pop);
+            ret = tsk_population_table_add_row(
+                &self->populations, pop.metadata, pop.metadata_length);
+            if (ret < 0) {
+                goto out;
+            }
+            population_map[node.population] = ret;
+        }
+        new_pop = population_map[node.population];
+    }
+    ret = tsk_node_table_add_row(&self->nodes, node.flags, node.time, new_pop,
```

Comment:
this should be something besides `ret`, since you only want to return the value if it's less than zero

Reply:
right! same for the add pop and add ind rows

Reply:
Gee, actually `ret` gets set to nonnegative things a bunch of times above; maybe you just want to change the check for whether `ret != 0` to `ret < 0` below?